activiti-core/activiti-engine/src/test/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImplTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2020 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.engine.impl.persistence.entity;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.util.ArrayList;
+import java.util.Date;
+import org.activiti.engine.delegate.event.ActivitiEventDispatcher;
+import org.activiti.engine.impl.cfg.PerformanceSettings;
+import org.activiti.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.activiti.engine.impl.context.Context;
+import org.activiti.engine.impl.persistence.entity.data.ExecutionDataManager;
+import org.activiti.engine.runtime.Clock;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+public class ExecutionEntityManagerImplTest {
+
+    @InjectMocks
+    private ExecutionEntityManagerImpl executionEntityManager;
+
+    @Mock
+    private ExecutionDataManager executionDataManager;
+
+    @Mock
+    private ProcessEngineConfigurationImpl processEngineConfiguration;
+
+    @Mock
+    private ActivitiEventDispatcher eventDispatcher;
+
+    @Mock
+    private Clock clock;
+
+    @Before
+    public void setUp() throws Exception {
+        initMocks(this);
+        given(processEngineConfiguration.getClock()).willReturn(clock);
+        given(processEngineConfiguration.getEventDispatcher()).willReturn(eventDispatcher);
+        Context.setProcessEngineConfiguration(processEngineConfiguration);
+    }
+
+    @Test
+    public void should_createProcessInstanceExecution_when_NoInitialVariable() {
```

Comment:
i don't understand the method name -> "NoInitialVariable" What variable does it refer to?

Reply:
`when_initialVariableNull` is better?

Reply:
basically I test the case when the on the processInstance creation we are not setting any variable

Reply:
when_initialVariableNameIsNull ? 
I would like to add "Name" as it is in the method signature
https://github.com/Activiti/Activiti/blob/fc2793f5f896c9d9b0acdb52cf44904b67e882b1/activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/persistence/entity/ExecutionEntityManagerImpl.java#L210

Reply:
is there a special behaviour triggered when initialVariableName is null?